src/main/java/com/snowflake/kafka/connector/internal/FileNameUtils.java

```diff
@@ -192,4 +196,39 @@ private static String readFromFileName(String fileName, int index) {
 
     return matcher.group(index);
   }
+
+  /**
+   * Find gaps in offset ranges.
+   *
+   * @param filenames list of files
+   * @return continuous and missing offsets for given filenames
+   */
+  static OffsetContinuityRanges searchForMissingOffsets(List<String> filenames) {
+    List<Pair<Long, Long>> missingOffsets = new ArrayList<>();
+
+    List<Pair<Long, Long>> continuousOffsets =
```

Comment:
i think we should also sort it, right?

Reply:
Good point, I made an assumption that they are already sorted.